features/netsocket/NetworkInterface.h

```diff
@@ -378,6 +378,15 @@ class NetworkInterface: public DNS {
      */
     virtual nsapi_error_t set_blocking(bool blocking);
 
+    /** Get suggested data transmission jitter from underlying network interface.
+     *
+     * @param data_amount Amount of data to be sent
+     * @param jitter Address to location where suggested jitter value is written
+     * @return NSAPI_ERROR_OK on success
+     * @return NSAPI_ERROR_UNSUPPORTED, NSAPI_ERROR_NO_CONNECTION
+     */
+    virtual nsapi_error_t get_interface_property(unsigned int data_amount, unsigned int *jitter);
```

Comment:
Why is it needed in top level?

Reply:
Having API here allows application to be independent of actual interface.

Reply:
The ROM cost here due to the virtual methods, particularly if broken up, is significant. And these are quite specialised calls.

Therefore my inclination would be to do this via new reason codes for `getsockopt`, avoiding adding anything to the vtable.

Particularly as something like RTT information may in principle be connection/remote-peer-specific, not simply a property of the interface, and the application ultimately is going to want to be working with a socket anyway.

Note that you can still have "nice wrappers" around a setsockopt, such as `InternetSocket::join_multicast_group` or `TLSSocket::set_hostname` (offboard implementation). They would be non-virtual, so no overhead.